### PR TITLE
Add a regression test for the modules used from the JDK.

### PR DESCRIPTION
**Pull request by meisterT**
It's critical for Bazel's binary size that we don't increase the list of
modules.
We will also use the same list in a later commit build a reduced JDK.

### CODE REVIEW

**Comment by buchgr** [comment]
Does the test fail if you remove one of the modules? :stuck_out_tongue:

File: src/test/shell/bazel/BUILD [hunk 1]
```diff
@@ -737,3 +737,18 @@ sh_test(
...
+
+sh_test(
+    name = "jdeps_test",
+    size = "medium",
+    srcs = ["jdeps_test.sh"],
+    data = [
+        ":jdeps_modules.golden",
+        ":jdeps_class_blacklist.txt",
+        ":test-deps",
+        "//src:embedded_jdk",
```
Comment (buchgr) [inline src/test/shell/bazel/BUILD]: Once we minimize the jdk it will not contain a `jdeps` tool?

**Comment by meisterT** [comment]
> Does the test fail if you remove one of the modules?

Yes, I described what to do in which failure case in the header of the test.

File: src/test/shell/bazel/BUILD (hunk 1 above)
Comment (meisterT) [reply src/test/shell/bazel/BUILD]: The dependency here will change (probably just the name). So this test will depend on the allmodules JDK.

**Comment by meisterT** [comment]
Passes the test on Mac as well now.

**Pull request closed without merging**